### PR TITLE
new radio widget, stories, changes for controls

### PR DESCRIPTION
**Pull request by moontrip**
This is required for the change from box button to radio button for scatterplot viz. This PR includes several works:

- created radio button group widget with various props and made its story
- changed XYPlotControls with the new widget and its story
- changed useControls with the name of XYPlot
- clean-up barplot and its story

A screenshot is attached:

radiowidget1

### CODE REVIEW

**Comment by moontrip** [comment]
a new commit is added to allow barplot's width/height props to accept both number and height values. This will be used at barplot viz as well

File: src/components/widgets/RadioButtonGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  labelPlacement,
+  minWidth,
+  buttonColor,
+  setMargin,
+}: RadioButtonGroupProps) {
+  // perhaps not using focused?
+  // const [focused, setFocused] = useState(false);
+
+  //DKDK
+  console.log('selectedOption = ', selectedOption);
```
Comment (jtlong3rd) [inline src/components/widgets/RadioButtonGroup.tsx:53]: Can we delete these `console.log`s?

PS: Out of curiosity, have you tried [React DevTools](https://www.npmjs.com/package/react-devtools)? I find it very helpful for keeping track of prop, state, and other hook values.

File: src/components/plotControls/XYPlotControls.tsx [hunk 2]
```diff
@@ -24,16 +23,26 @@ export type XYPlotControlsProps = {
...
+  /** How buttons are displayed. Vertical or Horizontal */
+  orientation?: 'vertical' | 'horizontal';
+  /** location of radio button label: start: label & button; end: button & label */
+  labelPlacement?: 'start' | 'end' | 'top' | 'bottom';
+  /** minimum width to set up equivalently spaced width per item */
+  minWidth?: number;
+  /** button color: for now, supporting blue and red only - primary: blue; secondary: red */
+  buttonColor?: 'primary' | 'secondary';
+  /** margin of radio button group: string array for top, left, bottom, and left, e.g., ['10em', '0', '0', '10em'] */
+  setMargin?: string[];
```
Comment (jtlong3rd) [inline src/components/plotControls/XYPlotControls.tsx:41]: I would name this prop something like `margins`. The name `setMargin` would make more sense if the prop were some sort of callback.

**Review by jtlong3rd — CHANGES_REQUESTED** [review]
This looks great! I had two minor suggestions.

File: src/components/widgets/RadioButtonGroup.tsx (hunk 1 above)
Comment (moontrip) [reply src/components/widgets/RadioButtonGroup.tsx:53]: @jtlong3rd First of all, thank you for your review 👍 For that console.log(), usually I tend to clean up those stuffs after approval, as they may need more works after review, but anyway I will remove them.

Yes, I am using react devtools definitely, but yeah it is a bit old habit to use console.log() :)

File: src/components/plotControls/XYPlotControls.tsx (hunk 2 above)
Comment (moontrip) [reply src/components/plotControls/XYPlotControls.tsx:41]: @jtlong3rd Actually I was gauging how to name that props: `margin/margins/setMargin/setMargins` haha. Certainly `setMargin` is more like callback stuff as you said, so will change it to `margins`

File: src/components/widgets/RadioButtonGroup.tsx (hunk 1 above)
Comment (jtlong3rd) [reply src/components/widgets/RadioButtonGroup.tsx:53]: No worries - it's an old habit for me too! 😆

**Review by bobular — APPROVED** [review]
Approve, after taking into consideration Jamie and my suggestions about the margin prop.

Thanks

**Review by jtlong3rd — APPROVED** [review]

**Comment by moontrip** [comment]
I have discussed with Bob regarding the spacing one. We agree that there needs to be major tidy-up of all the plot props, so for the time being I will use margins props as is and then revisit with more standardized props later. Will merge this.

Thank you for your reviews, @bobular and @jtlong3rd 👍

**Pull request merged**